tests/test_bgt.py

```diff
@@ -13,96 +14,64 @@
 
 # local
 from app._version import __version__
-from app.schema import Game
+from app.schema import Game, Question
 from app.utils import UniversalEncoder
 
 
-@fixture
-def aws_credentials():
-    """Mocked AWS Credentials for moto."""
-    os.environ['AWS_ACCESS_KEY_ID'] = 'testing'
-    os.environ['AWS_SECRET_ACCESS_KEY'] = 'testing'
-    os.environ['AWS_SECURITY_TOKEN'] = 'testing'
-    os.environ['AWS_SESSION_TOKEN'] = 'testing'
-
-
-@fixture
-def games_data():
-    # load game data
-    with open(Path(__file__).parent/"games_test_data.json", "r") as games_put_json:
-        games_raw = load(games_put_json)
-        # validate/cast game data
-    games = []
-    for game in games_raw:
-        games.append(Game(**game))
-    return games
-
-
-@fixture
-def outside_game_data():
-    with open(Path(__file__).parent/"outside_game.json", "r") as outside_game_json:
-        return Game(**load(outside_game_json))
-
-
-@fixture
-def setup_db(games_data, aws_credentials):
-    with mock_dynamodb2():
-        # load table schema
-        with open(Path(__file__).parent.parent/"games_table_schema.json", "r") as schema_json:
-            schema = load(schema_json)
-
-        # create dynamodb interface (mocked by moto decorator)
-        dynamodb = boto3.resource('dynamodb', 'us-east-1')
-
-        # create table with schema
-        table = dynamodb.create_table(
-            **schema
-        )
-        table.wait_until_exists()
-
-        # put games in table
-        with table.batch_writer() as batch:
-            for game in games_data:
-                batch.put_item(game.dict())
-        from app._main import app
-        client = TestClient(app)
-        yield client
-
-
 def test_version():
-    assert __version__ == '0.1.0'
+    with open(Path(__file__).parent.parent / "pyproject.toml") as f:
+        pyproject_toml = toml.load(f)
+    assert __version__ == pyproject_toml["tool"]["poetry"]["version"]
 
 
-def test_get_nogame(outside_game_data, setup_db):
+def test_get_nogame(outside_game_data, client):
     "Makes sure we're testing against the mocked DB and not the real one"
-    client = setup_db
     test_game = outside_game_data
     assert hasattr(test_game, "id")

```

Comment:
Very generally speaking, you don't often want to "test your tests." I'd just make sure everything is configured properly for your tests to run the way you want them to, and leave the rest up to your continuous development cycle.

Reply:
Gotcha. I initially did this to see if I was mocking my db correctly, and it showed my that I wasn't T_T lol

Generally, is it reasonable to "test your tests" initially and then eliminate them? I suppose this was more of a learning step for me, and the opacity of moto and boto3, then a typical development step... have you ever had a situation where you weren't sure if you were using a tool correctly? How would you go about making sure you've done it right?

Reply:
Yeah, I'll occasionally do something like that, but typically won't include that test in the commit. Alternatively, you could create a separate, reusable package for test utilities, and now that you've defined some new, reusable units, they're starting to become worthy of unit tests.

Generally, I'd either do what you did, by running and then deleting a temporary test, or I'd open up the Python REPL and play around with things live.

There is another useful class of test (integration test) where you can assert various things about the behavior of your dependencies. They're pretty helpful if you want to give yourself some guarantee about how a dependency works as you upgrade it. I don't rely on these too heavily. Every once in a while I find myself in a situation where I have to make an unfair assumption about a library in order to extend it for my uses (ie access or manipulate a private member). Because it's so unsafe, I will almost always accompany such a change with an integration test to prove my assumptions about the code still stand as I upgrade the library.